### DISCUSSION TITLE
EfficientDet D0 input image size

### DISCUSSION THREAD

**Discussion by Ekta246** [rwightman/efficientdet-pytorch#131] [Q&A]
@MichaelMonashev I hope you can help me with this.
I read the issue on non square anchor layout.

Correct me if I am wrong.
The input image size for the EfficientDet-D0 should be (512, 512) irrespective of any input image size of the custom dataset right?

In my assumption the transforms.py resizes and pads to 512 even if we just give the target['img_size'] param and not modifying the 512, 512 size in the config.py for EfficientDet D0

Whereas, for the target['img_size'] param in the dictionary should we input the original image size? Or we should modify the existing model_config for the D0 model for the target['img_size'] param?

In my understanding, the model_config input image size should not be changed since EffDet-D0 requires 512 while the D1 requires 640 and so on.
So where should we put the non square input original image size for the dataset?

**Comment by MichaelMonashev** [comment]
I tried using non square anchor layout and got very strange bboxes, so I padd all images to square.

**Reply by Ekta246** [reply]
Yes, I pad all the images to square(which is taken care by the transforms RandomResizePad I believe) and see pretty good results.
But, I still pass the target['image_size'] in the target dict as the non square image size and see good improvements over the predictions.

**Comment by Ekta246** [comment]
> I tried using non square anchor layout and got very strange bboxes, so I padd all images to square.

Thanks for the answer!
Also, do you change the model_config.py default img_size (512, 512) for say, efficientdet-d0 to your original image size? Or you just mention the img_size in the target['img_size']?

The confusion has raised due to the efficientdet-d0 model needs (512,512) image_size.
